test/source-updater.test.js

```diff
@@ -1,232 +1,769 @@
-import SourceUpdater from '../src/source-updater';
+import document from 'global/document';
+import window from 'global/window';
 import QUnit from 'qunit';
 import videojs from 'video.js';
-import { useFakeMediaSource } from './test-helpers';
+import SourceUpdater from '../src/source-updater';
+import { mp4Video, mp4Audio } from './test-segments';
 
 QUnit.module('Source Updater', {
   beforeEach() {
-    this.mse = useFakeMediaSource();
-    this.mediaSource = new videojs.MediaSource();
-  },
-  afterEach() {
-    this.mse.restore();
+    const video = document.createElement('video');
+
+    this.mediaSource = new window.MediaSource();
+    // need to attach the real media source to a video element for the media source to
+    // change to an open ready state
+    video.src = URL.createObjectURL(this.mediaSource);
+    this.sourceUpdater = new SourceUpdater(this.mediaSource);
+
+    // wait for the source to open (or error) before running through tests
+    return new Promise((accept, reject) => {
```

Comment:
Each test should wait to start until that promise resolves. And if it does reject then it's good that it'll fail (since the test probably can't run without an open media source).